app/Helpers/database/game.php

```diff
@@ -910,40 +920,36 @@ function submitNewGameTitleJSON(
     return $retVal;
 }
 
-function modifyGameRichPresence(string $user, int $gameID, string $dataIn): bool
+function modifyGameRichPresence(string $username, int $gameId, string $dataIn): bool
 {
-    getRichPresencePatch($gameID, $existingData);
+    getRichPresencePatch($gameId, $existingData);
     if ($existingData == $dataIn) {
         return true;
     }
 
-    sanitize_sql_inputs($dataIn);
-    $query = "UPDATE GameData SET RichPresencePatch='$dataIn' WHERE ID=$gameID";
-
-    $db = getMysqliConnection();
-    $dbResult = mysqli_query($db, $query);
-    if (!$dbResult) {
+    $game = Game::find($gameId);
+    if (!$game) {
         return false;
     }
 
-    addArticleComment('Server', ArticleType::GameModification, $gameID, "$user changed the rich presence script");
+    $game->RichPresencePatch = $dataIn;
+    $game->save();
+
+    addArticleComment('Server', ArticleType::GameModification, $gameId, "{$username} changed the rich presence script");
 
     return true;
 }
 
-function getRichPresencePatch(int $gameID, ?array &$dataOut): bool
+function getRichPresencePatch(int $gameId, ?array &$dataOut): bool
```

Comment:
$dataOut should be a string.

Reply:
Changed to string in latest. This one escaped me - has this been incorrectly typed the whole time?